test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -2335,3 +2341,8 @@ def forward(self, x):
 
 
 instantiate_parametrized_tests(TestQuantizePT2E)
+
+devices = ["cpu", "cuda"]
+if TEST_HPU:
+    devices.append("hpu")
+instantiate_device_type_tests(TestQuantizePT2E, globals(), only_for=devices)
```

Comment:
cc @amathewc this seems to break the import of TestQuantizePT2E, did you run test for the change?
```
WARNING:root:cannot import name 'TestQuantizePT2E' from 'quantization.pt2e.test_quantize_pt2e' (/data/users/jerryzh/pytorch/test/quantization/pt2e/test_quantize_pt2e.py)
```

Reply:
@jerryzh168 : Since we are using instantiate_device_type_tests , the test will get  instantiated for each device in the system. 
Launching the tests with # pytest -vs test_quantize_pt2e.py (without specifying a class name ) will launch the tests on all devices. 
If you want the tests to be run 
- specifically on CPUs : pytest -vs test_quantize_pt2e.py::TestQuantizePT2eCPU
- specifically on CUDA devices: pytest -vs test_quantize_pt2e.py::TestQuantizePT2eCUDA
- specifically on HPUs: pytest -vs test_quantize_pt2e.py::TestQuantizePT2eHPU

/pytorch/test/quantization/pt2e# pytest  test_quantize_pt2e.py --collect-only
==================================================================================================== test session starts =====================================================================================================
platform linux -- Python 3.11.10, pytest-8.3.3, pluggy-1.5.0
rootdir: /home/pytorch
configfile: pytest.ini
plugins: hypothesis-6.115.3
collected 146 items
Running 146 items in this shard

<Dir pytorch>
  <Dir test>
    <Package quantization>
      <Dir pt2e>
        <Module test_quantize_pt2e.py>
          <UnitTestCase TestQuantizePT2ECPU>
            <TestCaseFunction test_allow_exported_model_train_eval_cpu>
            <TestCaseFunction test_allow_implicit_sharing_cpu>
            <TestCaseFunction test_composable_quantizer_linear_conv_cpu>
            <TestCaseFunction test_composable_quantizer_throw_cpu>
            <TestCaseFunction test_composable_quantizer_transform_for_annotation_cpu>
            <TestCaseFunction test_constant_prop_preserve_metadata_cpu>
            <TestCaseFunction test_conv_transpose_bn_relu_cpu>
            <TestCaseFunction test_derived_qspec_cpu>
            <TestCaseFunction test_derived_qspec_per_channel_cpu>
            <TestCaseFunction test_disallow_eval_train_cpu>
            <TestCaseFunction test_dont_fold_other_constant_cpu>
            <TestCaseFunction test_embedding_conv_linear_quantization_cpu>
            <TestCaseFunction test_embedding_quantizer_cpu>
            <TestCaseFunction test_fixed_qparams_qspec_observer_dedup_cpu>
            <TestCaseFunction test_fixed_qparams_qspec_ptq_cpu>
            <TestCaseFunction test_fixed_qparams_qspec_qat_cpu>
            <TestCaseFunction test_fold_all_ops_before_quantize_cpu>
            <TestCaseFunction test_fold_quantize_cpu>
            <TestCaseFunction test_fold_quantize_per_channel_cpu>
            <TestCaseFunction test_groupwise_per_channel_quant_cpu>
            <TestCaseFunction test_input_edge_sanity_check_cpu>
            <TestCaseFunction test_max_pool2d_quantizer_cpu>
            <TestCaseFunction test_model_is_exported_cpu>
            <TestCaseFunction test_move_exported_model_bn_cpu>
            <TestCaseFunction test_move_exported_model_dropout_cpu>
            <TestCaseFunction test_move_exported_model_dropout_inplace_cpu>
            <TestCaseFunction test_multi_users_without_output_observer_cpu>
            <TestCaseFunction test_observer_callback_cpu>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_bfloat16_float8_e4m3fn_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_bfloat16_float8_e5m2_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_bfloat16_int16_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_float32_float8_e4m3fn_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_float32_float8_e5m2_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_float32_int16_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_bfloat16_float8_e4m3fn_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_bfloat16_float8_e5m2_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_bfloat16_int16_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_float32_float8_e4m3fn_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_float32_float8_e5m2_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_float32_int16_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_bfloat16_float8_e4m3fn_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_bfloat16_float8_e5m2_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_bfloat16_int16_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_float32_float8_e4m3fn_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_float32_float8_e5m2_cpu_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_float32_int16_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_bfloat16_float8_e4m3fn_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_bfloat16_float8_e5m2_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_bfloat16_int16_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_float32_float8_e4m3fn_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_float32_float8_e5m2_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_float32_int16_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_bfloat16_float8_e4m3fn_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_bfloat16_float8_e5m2_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_bfloat16_int16_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_float32_float8_e4m3fn_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_float32_float8_e5m2_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_float32_int16_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_int16_bfloat16_float8_e4m3fn_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_int16_bfloat16_float8_e5m2_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_int16_bfloat16_int16_cpu_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_int16_float32_float8_e4m3fn_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_int16_float32_float8_e5m2_cpu_float32>
            <TestCaseFunction test_quantization_dtype_float32_int16_float32_int16_cpu_float32>
            <TestCaseFunction test_reentrant_cpu>
            <TestCaseFunction test_save_load_cpu>
            <TestCaseFunction test_shared_qspec_cpu>
            <TestCaseFunction test_shared_qspec_transitivity_case_2_cpu>
            <TestCaseFunction test_shared_qspec_transitivity_cpu>
            <TestCaseFunction test_simple_quantizer_cpu>
            <TestCaseFunction test_speed_cpu>
            <TestCaseFunction test_transform_for_annotation_cpu>
            <TestCaseFunction test_wo_annotate_conv_output_quantizer_cpu>
          <UnitTestCase TestQuantizePT2ECUDA>
            <TestCaseFunction test_allow_exported_model_train_eval_cuda>
            <TestCaseFunction test_allow_implicit_sharing_cuda>
            <TestCaseFunction test_composable_quantizer_linear_conv_cuda>
            <TestCaseFunction test_composable_quantizer_throw_cuda>
            <TestCaseFunction test_composable_quantizer_transform_for_annotation_cuda>
            <TestCaseFunction test_constant_prop_preserve_metadata_cuda>
            <TestCaseFunction test_conv_transpose_bn_relu_cuda>
            <TestCaseFunction test_derived_qspec_cuda>
            <TestCaseFunction test_derived_qspec_per_channel_cuda>
            <TestCaseFunction test_disallow_eval_train_cuda>
            <TestCaseFunction test_dont_fold_other_constant_cuda>
            <TestCaseFunction test_embedding_conv_linear_quantization_cuda>
            <TestCaseFunction test_embedding_quantizer_cuda>
            <TestCaseFunction test_fixed_qparams_qspec_observer_dedup_cuda>
            <TestCaseFunction test_fixed_qparams_qspec_ptq_cuda>
            <TestCaseFunction test_fixed_qparams_qspec_qat_cuda>
            <TestCaseFunction test_fold_all_ops_before_quantize_cuda>
            <TestCaseFunction test_fold_quantize_cuda>
            <TestCaseFunction test_fold_quantize_per_channel_cuda>
            <TestCaseFunction test_groupwise_per_channel_quant_cuda>
            <TestCaseFunction test_input_edge_sanity_check_cuda>
            <TestCaseFunction test_max_pool2d_quantizer_cuda>
            <TestCaseFunction test_model_is_exported_cuda>
            <TestCaseFunction test_move_exported_model_bn_cuda>
            <TestCaseFunction test_move_exported_model_dropout_cuda>
            <TestCaseFunction test_move_exported_model_dropout_inplace_cuda>
            <TestCaseFunction test_multi_users_without_output_observer_cuda>
            <TestCaseFunction test_observer_callback_cuda>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_bfloat16_float8_e4m3fn_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_bfloat16_float8_e5m2_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_bfloat16_int16_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_float32_float8_e4m3fn_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_float32_float8_e5m2_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e4m3fn_float32_int16_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_bfloat16_float8_e4m3fn_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_bfloat16_float8_e5m2_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_bfloat16_int16_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_float32_float8_e4m3fn_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_float32_float8_e5m2_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_float8_e5m2_float32_int16_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_bfloat16_float8_e4m3fn_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_bfloat16_float8_e5m2_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_bfloat16_int16_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_float32_float8_e4m3fn_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_float32_float8_e5m2_cuda_float32>
            <TestCaseFunction test_quantization_dtype_bfloat16_int16_float32_int16_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_bfloat16_float8_e4m3fn_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_bfloat16_float8_e5m2_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_bfloat16_int16_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_float32_float8_e4m3fn_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_float32_float8_e5m2_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e4m3fn_float32_int16_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_bfloat16_float8_e4m3fn_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_bfloat16_float8_e5m2_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_bfloat16_int16_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_float32_float8_e4m3fn_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_float32_float8_e5m2_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_float8_e5m2_float32_int16_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_int16_bfloat16_float8_e4m3fn_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_int16_bfloat16_float8_e5m2_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_int16_bfloat16_int16_cuda_bfloat16>
            <TestCaseFunction test_quantization_dtype_float32_int16_float32_float8_e4m3fn_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_int16_float32_float8_e5m2_cuda_float32>
            <TestCaseFunction test_quantization_dtype_float32_int16_float32_int16_cuda_float32>
            <TestCaseFunction test_reentrant_cuda>
            <TestCaseFunction test_save_load_cuda>
            <TestCaseFunction test_shared_qspec_cuda>
            <TestCaseFunction test_shared_qspec_transitivity_case_2_cuda>
            <TestCaseFunction test_shared_qspec_transitivity_cuda>
            <TestCaseFunction test_simple_quantizer_cuda>
            <TestCaseFunction test_speed_cuda>
            <TestCaseFunction test_transform_for_annotation_cuda>
            <TestCaseFunction test_wo_annotate_conv_output_quantizer_cuda>

================================================================================================ 146 tests collected in 3.39s ================================================================================================

Reply:
I see, thanks, but we are using `python test/test_quantization.py -k test_name` e.g. `python test/test_quantization.py -k test_simple_quantizer` to run all the other tests, so this should work, can you check how to make this work?